core/include/gen-asm-defines.h

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
I don't think it's worth having a separate file for this, could you move these definitions into `core/arch/arm/kernel/asm-defines.c` instead?